pkg/apis/cluster/validation/validation.go

```diff
@@ -148,6 +148,30 @@ func ValidateClusterProxyURL(fldPath *field.Path, proxyURL string) field.ErrorLi
 	return allErrs
 }
 
+func validateClusterSpecZoneLabel(spec *api.ClusterSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	// Zone and Zones cannot co-exist.
+	// see https://github.com/karmada-io/karmada/issues/3952
+	if spec.Zone != "" && len(spec.Zones) != 0 {
+		return append(allErrs, field.TypeInvalid(fldPath.Child("spec"), spec, "Zone and Zones cannot co-exist"))
+	}
+
+	if spec.Zone != "" {
```

Comment:
we don't need to validate `.spec.zone` because it will be filled to `.spec.zones` and remain empty.

Reply:
It doesn't harm though. Both are okay to me.

Reply:
OK,I'll keep it.
And here will be changed later when the zone field is removed.